evm-adapters/src/sputnik/forked_backend.rs

```diff
@@ -21,7 +27,7 @@ pub struct ForkMemoryBackend<B, M> {
     // TODO: Actually cache values in memory.
     // TODO: This should probably be abstracted away into something that efficiently
     // also caches at disk etc.
-    pub cache: BTreeMap<H160, MemoryAccount>,
+    pub cache: RefCell<BTreeMap<H160, MemoryAccount>>,
```

Comment:
this could be its own struct/trait which abstracts all reading writing, something like:

```rust
#[derive(Clone)]
pub struct Cache<Value> {
    inner: Arc<RwLock<InnerCache<T>>,
}

struct InnerCache<T> {
    cached: HashMap<String, CachedValue<T>>,
}

Since the values are retrieved via async, we probably need something like this to keep track if we're currently fetching some value

pub enum CachedValue<Value> {
    Cached(Value),
    Pending(<Maybe the future that returns the Value>),
}
```
Perhaps this can be simplified with `lru::Cache` or the `cached` crate which supports async

Reply:
Opened https://github.com/gakonst/dapptools-rs/issues/71 to track